File: 7-kyu/sum-of-odd-numbers/sum-of-odd-numbers.go
package kata
​
func RowSumOddNumbers(n int) int {
  floorNum := (n - 1) * (n) / 2; // triangular numbers' formula
  ceilNum := (n) * (n + 1) / 2;
  sum := 0;
  for floorNum < ceilNum {
    sum += floorNum * 2 + 1;
    floorNum++;
  }
  return sum;
}
​

File: 7-kyu/sum-of-odd-numbers/sum-of-odd-numbers.java
class RowSumOddNumbers {
    public static int rowSumOddNumbers(int n) {
      int floorNum = (n - 1) * (n) / 2;
      int ceilNum = (n) * (n + 1) / 2;
      int sum = 0;
      while (floorNum < ceilNum) {
        sum += (floorNum * 2) + 1;
        floorNum++;
      }
      return sum;
    }
}